oolab/src/main/java/agh/ics/oop/MapBoundary.java

```diff
@@ -0,0 +1,38 @@
+package agh.ics.oop;
+
+import java.util.*;
+
+public class MapBoundary implements IPositionChangeObserver{
+    private Vector2d position;
+    private final Map<Vector2d, classTypeAndPositionTuple> objects = new HashMap<>();
+    private final SortedSet<classTypeAndPositionTuple> objectsX = new TreeSet<>(classTypeAndPositionTuple.compareToByX);
+    private final SortedSet<classTypeAndPositionTuple> objectsY = new TreeSet<>(classTypeAndPositionTuple.compareToByY);
+
+    public MapBoundary(){
+        return;
+    }
+
+    public void place(MapObject object){
+        objects.put(object.position, new classTypeAndPositionTuple(object.position, object.getClass()));
+        objectsX.add(new classTypeAndPositionTuple(object.position, object.getClass()));
+        objectsY.add(new classTypeAndPositionTuple(object.position, object.getClass()));
+        if(object instanceof Animal){
```

Comment:
ładniej by było gdyby mapa przy dodawaniu zwierzątka ustawiała tego observera, mniej instanceofów to mniej komplikacji.